### ISSUE TITLE
Implement device auth

### ISSUE THREAD

**Issue by Razzeee** [fuzeman/trakt.py#46]
See https://plus.google.com/109991097847241028373/posts/c143ezzcZ8R

http://docs.trakt.apiary.io/#reference/authentication-devices

**Comment by fuzeman** [comment]
Will look into this next week :smile:

**Comment by Razzeee** [comment]
No hurry

**Comment by fuzeman** [comment]
Early implementation is available on the "develop" branch now, just need to write some more unit tests and ensure there isn't any bugs with it.

**Comment by fuzeman** [comment]
Unit tests have been added, it will probably need some real-world testing to spot any other issues :smile:

Example implementation is available here: https://github.com/fuzeman/trakt.py/blob/develop/examples/authentication/device.py

**Comment by Razzeee** [comment]
Thank you! I will have a go at this soon.

**Comment by Razzeee** [comment]
Seems to work fine https://github.com/Razzeee/script.trakt/commits/device-auth

Is there any plan for a stable release?

**Comment by fuzeman** [comment]
Released in [v2.6.0](https://github.com/fuzeman/trakt.py/releases/tag/2.6.0)

**Comment by Razzeee** [comment]
Do we still need to refresh tokens? Got a user report with following log entry
```
WARNING: [script.trakt] trakt.core.http: OAuth - Unable to refresh expired token (token refreshing hasn't been enabled)
```

**Comment by fuzeman** [comment]
> Do we still need to refresh tokens?

Yes, all authentication tokens have an expiry. You should be able to use the same token refresh handling from the pin authentication method.

Just noticed this was missing from the device authentication example, will get that updated soon.

**Comment by Razzeee** [comment]
Yep, I build everything on the example so this is missing in my implementation

**Comment by Razzeee** [comment]
@fuzeman like that?
https://github.com/Razzeee/script.trakt/commit/00e982c2fe74d420a0e18d794e59b4ef22206f84

**Comment by fuzeman** [comment]
@Razzeee Yeah, that looks good. :+1: 

I've also fixed an issue where tokens generated with device authentication wouldn't refresh correctly *(as there was no `created_at` parameter returned from the Trakt.tv API)*, will release a new version of trakt.py with this fix later today.

**Comment by fuzeman** [comment]
Just released [v2.6.1](https://github.com/fuzeman/trakt.py/releases/tag/2.6.1).

This release fixes an issue where tokens generated with device authentication didn't have a `created_at` parameter, this meant token expiration checks were being disabled and tokens never being refreshed.
